arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,1837 @@
+      "required": [
+        "serviceUri"
+      ],
+      "properties": {
+        "serviceUri": {
+          "description": "The service URI for the webhook to post notifications.",
+          "type": "string"
+        },
+        "customHeaders": {
+          "description": "Custom headers that will be added to the webhook notifications.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "WebhookUpdateParameters": {
+      "description": "The parameters for updating a webhook.",
+      "properties": {
+        "tags": {
+          "description": "The tags for the webhook.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/WebhookPropertiesUpdateParameters",
+          "description": "The properties that the webhook will be updated with.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "WebhookPropertiesUpdateParameters": {
+      "description": "The parameters for updating the properties of a webhook.",
+      "required": [
+        "callbackConfig"
+      ],
+      "properties": {
+        "callbackConfig": {
+          "$ref": "#/definitions/CallbackConfig",
+          "description": "The configuration of service URI and custom headers for the webhook."
+        },
+        "isEnabled": {
+          "description": "The value that indicates whether the webhook is enabled.",
+          "type": "boolean"
+        },
+        "scope": {
+          "description": "The scope of repositories where the event can be triggered.",
+          "type": "string"
+        },
+        "actions": {
+          "description": "The list of actions that trigger the webhook to post notifications.",
+          "type": "array",
+          "items": {
+            "enum": [
+              "push",
+              "pull",
+              "delete"
+            ],
+            "type": "string",
+            "x-ms-enum": {
+              "name": "WebhookAction",
+              "modelAsString": true
+            }
+          }
+        }
+      }
+    },
+    "WebhookListResult": {
+      "description": "The result of a request to list webhooks for a container registry.",
+      "properties": {
+        "value": {
+          "description": "The list of webhooks. Since this list may be incomplete, the nextLink field should be used to request the next list of webhooks.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Webhook"
+          }
+        },
+        "nextLink": {
+          "description": "The URI that can be used to request the next list of webhooks.",
+          "type": "string"
+        }
+      }
+    },
+    "EventInfo": {
+      "description": "The basic information of an event.",
+      "properties": {
+        "id": {
```

Comment:
This is also a property only in response. If I add readonly here I feel obliged to add readonly to all properties under Event object.